src/libraries/Microsoft.Extensions.DependencyInjection/Directory.Build.props

```diff
@@ -0,0 +1,7 @@
+﻿<Project>
+  <Import Project="..\Directory.Build.props" />
+  <PropertyGroup>
+    <PackageVersion>5.0.1</PackageVersion>
+    <AssemblyVersion>5.0.0.1</AssemblyVersion>
```

Comment:
@dougbu @Anipik remind me what our plan is for references shared in the transport package?  Did we agree to freeze the assembly version or no?  If so, then we should move the AssemblyVersion change to the src project.

Reply:
@ericstj we agreed dotnet/runtime shouldn't create the transport package unless we _really_ need to service the targeting packs. I don't remember if @Pilchie's document specifically covers `$(AssemblyVersion)` for ref/ assemblies but do recall it recommends not changing `$(AssemblyVersion)` for anything but Framework assemblies in servicing.

Reply:
I believe in that document we concluded that assemblyVersion should change only apply to the net4x targetframeworks when an assembly overlaps the shared framework.  We could follow that rule here.  @Anipik did you have anything in the servicing doc about this with respect to ensuring servicable ref packs?

Reply:
> we concluded that assemblyVersion should change only apply to the net4x targetframeworks when an assembly overlaps the shared framework

yes we increment if we have any asset that applies netfx or netstandard, following the same thing here.

>  @Anipik did you have anything in the servicing doc about this with respect to ensuring servicable ref packs?

sadly nothing in the servicing doc with respect to ref packs. We need to add few more templates as well, i can add ref pack specific guidelines with that

Reply:
>> we concluded that assemblyVersion should change only apply to the net4x targetframeworks when an assembly overlaps the shared framework

> yes we increment if we have any asset that applies netfx or netstandard, following the same thing here.

@Anipik this doesn't seem to match the condition @ericstj described. NetStandard assemblies from projects that end up in the shared framework shouldn't vary because it ties external libraries to specific shared framework versions.

The exact `Condition` for stable versions (overriding Arcade's defaults) in dotnet/aspnetcore is `'$(IsAspNetCoreApp)' == 'true' AND '$(Language)' == 'C#' AND '$(TargetFrameworkIdentifier)' != '.NETFramework'`. This (plus targeting or multi-targeting `net5.0` in those projects) means all assemblies in Microsoft.AspNetCore.App from our repo have stable versions and the same is true of separately-shipped NetStandard assemblies from those projects. Everything else varies in servicing.

Crucially this is _not_ what dotnet/runtime is doing for assemblies that end up in Microsoft.AspNetCore.App. So far, Microsoft.Extensions.DependencyInjection and System.IO.Pipelines have new assembly versions. I needed to workaround the first change and revert bfc1ec679202 / #25851 (to handle the second) in #28087. bfc1ec679202 was our change in expectation of stable assembly versions from dotnet/runtime.